tests/utils/azure_users_test.py

```diff
@@ -228,3 +242,26 @@ def test_remove_user_from_group(
     get_group_id_by_group_name_mock.assert_called_once()
     get_user_mock.assert_called_once()
     assert expected_value == test_user
+
+
+@patch('utils.azure_users.AzureConnection.get_groups_and_users')
+@patch('requests.get')
+def test_users_functions_should_returns_all_users(
+    get_mock, get_groups_and_users_mock
+):
+    first_response = Response()
+    first_response.status_code = 200
+    first_response._content = (
+        b'{"odata.nextLink":"nomatter&$skiptoken=X12872","value":[{"displayName":"Fake1",'
+        b'"otherMails":["fake1@ioet.com"],"objectId":"1"}]} '
+    )
+
+    second_response = copy.copy(first_response)
+    second_response._content = b'{"value":[{"displayName":"Fake2","otherMails":["fake2@ioet.com"],"objectId":"1"}]}'
+
+    get_mock.side_effect = [first_response, second_response]
+    get_groups_and_users_mock.return_value = []
+
+    users = AzureConnection().users()
+
+    assert len(users) == 2
```

Comment:
In my opinion, I think that an assert should be added to compare if the response's elements are the same as the fake.

`users == get_mock.side_efect`

Reply:
Resolved in an internal discussion